sdk/nodejs/tests/unwrap.spec.ts

```diff
@@ -152,52 +162,62 @@ describe("unwrap", () => {
         it("with single output", testResources(
             createOutput(3, r1, r2),
             3,
+            [r1, r2],
             [r1, r2]));
 
         it("inside array", testResources(
             [createOutput(3, r1, r2)],
             [3],
+            [r1, r2],
             [r1, r2]));
 
         it("inside multi array", testResources(
             [createOutput(1, r1, r2),createOutput(2, r2, r3)],
             [1, 2],
+            [r1, r2, r3],
             [r1, r2, r3]));
 
         it("inside nested array", testResources(
             [createOutput(1, r1, r2), createOutput(2, r2, r3), [createOutput(3, r5)]],
             [1, 2, [3]],
+            [r1, r2, r3, r5],
             [r1, r2, r3, r5]));
 
         it("inside object", testResources(
             { a: createOutput(3, r1, r2) },
             { a: 3 },
+            [r1, r2],
             [r1, r2]));
 
         it("inside multi object", testResources(
             { a: createOutput(1, r1, r2), b: createOutput(2, r2, r3) },
             { a: 1, b: 2 },
+            [r1, r2, r3],
             [r1, r2, r3]));
 
         it("inside nested object", testResources(
             { a: createOutput(1, r1, r2), b: createOutput(2, r2, r3), c: { d: createOutput(3, r5) } },
             { a: 1, b: 2, c: { d: 3 } },
+            [r1, r2, r3, r5],
             [r1, r2, r3, r5]));
 
         it("across inner promise", testResources(
             createOutput(Promise.resolve(3), r1, r2),
             3,
+            [r1, r2],
             [r1, r2]));
 
         describe("with unknowns", () => {
             it("across 'all' without unknowns", testResources(
                 all([Promise.resolve({ a: createOutput(unknown, r1, r2)}), Promise.resolve({ b: createOutput(unknown, r3, r4)})]),
                 undefined,
+                [],
                 [r1, r2, r3, r4]));
```

Comment:
these tests now validate both behaviors. the old behavior about what resources you coudl get synchronously from an output and what resources you now get asynchronously.